TensorFlow autolog automatic run ending (#2094)
mlflow/mlflow: docs/source/tracking.rst

| TensorFlow Core  | All ``tf.summary.scalar`` calls                        | --                                                       | --            | --                                                                                                                                               |
+------------------+--------------------------------------------------------+----------------------------------------------------------+---------------+--------------------------------------------------------------------------------------------------------------------------------------------------+

Note that autologging for ``tf.keras`` is handled by :py:func:`mlflow.tensorflow.autolog`, not :py:func:`mlflow.keras.autolog`. 

If no active run exists when `autolog()` captures data, MLflow will automatically create a run to log information to.
Once training ends via calls to `tf.estimator.train()`, `tf.keras.fit()`, `tf.keras.fit_generator()`, `keras.fit()` or `keras.fit_generator()`,

> or once `tf.estimator` models are exported via `tf.estimator.export_saved_model()` or `tf.estimator.export_savedmodel()`, MLflow will automatically end that run.

Reviewer: ```suggestion
or once `tf.estimator` models are exported via `tf.estimator.export_saved_model()`, MLflow will automatically end that run.
```

(`tf.estimator.export_savedmodel` was deprecated in the last TF 1.x release, so I don't think we need to mention it here)

> **Note**: When a run is automatically created by `mlflow.tensorflow.autolog`, a call to `tf.estimator.train()` will end the current run but will not log the trained model to that run.

Reviewer: Maybe we can reword this more concisely as:

```suggestion
**Note**: When using the `tf.Estimator()` API, `mlflow.tensorflow.autolog` will record training metrics & params passed to `train`. Autologging also logs the models saved by `tf.estimator.export_saved_model()` to MLflow.
```

> To track a model in the same run, you will need to call `tf.estimator.export_saved_model()`, which will start the most recent `autolog`-created run

Reviewer: ```suggestion
```

> and log the model there. The run will be automatically ended afterwards.

Reviewer: ```suggestion
```
